src/pkg_config_gen.rs

```diff
@@ -5,27 +5,31 @@ use crate::install::InstallPaths;
 use std::path::{Component, Path, PathBuf};
 
 fn canonicalize<P: AsRef<Path>>(path: P) -> String {
+    use core::fmt::Write;
+
     let mut separator = "";
-    let out = path
-        .as_ref()
-        .components()
-        .map(|p| match p {
+    let mut out = String::with_capacity(64);
+
+    for component in path.as_ref().components() {
+        let component_str = component.as_os_str().to_string_lossy();
+
+        match component {
             Component::RootDir => {
                 separator = "/";
-                String::new()
             }
-            Component::Prefix(_) => p.as_os_str().to_string_lossy().to_string(),
-            _ => {
-                let c = format!("{}{}", separator, p.as_os_str().to_string_lossy());
+            Component::Prefix(_) => {
+                write!(out, "{}", component_str).unwrap();
```

Comment:
a write to a `String` will only fail when the system runs out of memory. The rust standard library generally disregards OOM, and it seems reasonable to do here too.

Reply:
It shouldn't be a problem indeed.